### PR TITLE
Create resource classes for Connect, Connect S2I and MirrorMaker

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

Some time ago we created the class `KafkaResources` with methods for naming different resources related to Kafka clusters. This is among other currently used in tests. But we never moved forward with this for another custom Resources. This Pr does the same for Kafka Connect (including S2I), and Kafka Mirror Maker. Kafka Bridge will be done in other PR together with some tests using it. 

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectS2IResources.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Returns the name of the Kafka Connect S2I {@code ImageStream} for a {@code KafkaConnectS2I} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code KafkaConnectS2I} resource.
+     * @return The name of the corresponding Kafka Connect S2I {@code ImageStream}.
+     */
+    public static String sourceImageStreamName(String clusterName) {
+        return deploymentName(clusterName) + "-source";
+    }
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectS2IResources.java]: in the javadoc should we distinguish between target and source?

Comment (scholzj) [reply api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectS2IResources.java]: Good point. Should be fixed now. Thanks

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=KafkaST#testKafkaAndZookeeperScaleUpScaleDown,KafkaST#testLoadBalancerTls,AllNamespaceST#testKafkaInDifferentNsThanClusterOperator,MirrorMakerST#testMirrorMakerTlsAuthenticated,UserST#testUpdateUser

**Pull request merged**